test/DomUpdates-test.js

```diff
@@ -21,8 +67,55 @@ describe('DomUpdates', () => {
     expect(domUpdates).to.be.an.instanceOf(DomUpdates);
   });
 
-  it('Should call populateUserWidget correctly', () => {
-    domUpdates.populateUserWidget(currentUser, data, userRepo);
-    expect(domUpdates.populateUserWidget).to.have.been.called(1);
+  it('Should call getElementByID with the correct arguments', () => {
+    domUpdates.changeInnerTextID('user-stride-length', 'HELLO');
+
+    expect(document.getElementById).to.have.been.called.with('user-stride-length');
+  });
+
+  it('Should call call getElementById each time changeInnerTextID is called with populateUserWidget', () => {
+    domUpdates.populateUserWidget(currentUser, sleepRepo, activityRepo, mostRecentDate);
+
+    expect(document.getElementById).to.have.been.called(10);
+  });
+
+  it('Should call call getElementById each time changeInnerTextID is called with populateTodayInfo', () => {
+    domUpdates.populateTodayInfo(currentUser, sleepRepo, hydrationRepo, activityRepo, mostRecentDate);
+
+    expect(document.getElementById).to.have.been.called(7);
+  });
+
+  it('Should call call getElementById each for each cell in a column', () => {
+    const weekInformation = [
+      { date: '2019/06/15', amount: 38, unit: 'oz' },
+      { date: '2019/06/16', amount: 91, unit: 'oz' },
+      { date: '2019/06/17', amount: 96, unit: 'oz' },
+      { date: '2019/06/18', amount: 70, unit: 'oz' },
+      { date: '2019/06/19', amount: 76, unit: 'oz' },
+      { date: '2019/06/20', amount: 71, unit: 'oz' },
+      { date: '2019/06/21', amount: 27, unit: 'oz' },
+    ];
+
+    domUpdates.fillColumn('h', weekInformation);
+
+    expect(document.getElementById).to.have.been.called(7);
+  });
+
+  it('Should know the day of the week based on the date given', () => {
+    expect(domUpdates.getStringOfWeekday('2020/07/26')).to.eql('Sunday');
+    expect(domUpdates.getStringOfWeekday('2020/07/28')).to.eql('Tuesday');
+  });
+
+  it('Should call getElementById correctly once to create the friend-card', () => {
+    domUpdates.populateFriendsCard(currentUser, userRepo);
+
+    expect(document.getElementById).to.have.been.called(1);
+    expect(document.getElementById).to.have.been.called.with('friend-list');
+  });
+
+  it('Should call call getElementById each time changeInnerTextID is called with populateLeaderBoard', () => {
+    domUpdates.populateLeaderBoard(userRepo, sleepRepo, activityRepo, mostRecentDate, sleepRepo.sleepData);
+
+    expect(document.getElementById).to.have.been.called(7);
```

Comment:
If we have time should we have a test for anytime the getElementbyID method is called across all files?

Reply:
yeah I think if we have time that would be a great idea!